### PR TITLE
WIP: Integrate LOS vector in CSLC-S1 static layers product

### PR DESCRIPTION
**Pull request by vbrancat**
This PR integrates the LOS vector in the CSLC-S1 static layers product. Specifically, the PR removes the heading_angle layer and adds east_los_vector and north_los_vector.

TO DOs:

- [x] 1. Merge ISCE3 PR computing LOS vectors in Topo
- [x] 2. Test if the correct LOS vectors are returned for a couple of test bursts
- [x] 3. Remove `incidence_angle` and `heading`
- [x] 4. Check generated static layers product
- [ ] 5. Modify static layers product specifications

### CODE REVIEW

File: src/compass/s1_rdr2geo.py [hunk 1]
```diff
@@ -116,7 +116,12 @@ def run(cfg, burst=None, save_in_scratch=False):
                        rdr2geo_cfg.compute_local_incidence_angle,
                        gdal.GDT_Float32),
                        'heading': (
-                       rdr2geo_cfg.compute_azimuth_angle, gdal.GDT_Float32)}
+                       rdr2geo_cfg.compute_azimuth_angle, gdal.GDT_Float32),
```
Comment (LiangJYu) [inline src/compass/s1_rdr2geo.py:119]: Not related to this changes here - Can you change the comment above at 106 to something like [this](https://github.com/opera-adt/COMPASS/blob/e3518a9ac22b6461870dd88ecb917bc250c958ae/src/compass/s1_geocode_metadata.py#L130-L132). My not as nicely formatted commented below:
```
save output rasters to dict with key as topo layer file name and item as dict of bool to enable/disable and datatype
```
My original comment is pretty close to useless.

File: src/compass/s1_rdr2geo.py [hunk 2]
```diff
@@ -116,7 +116,12 @@ def run(cfg, burst=None, save_in_scratch=False):
...
+                       'los_east': (
+                       rdr2geo_cfg.compute_ground_to_sat_east, gdal.GDT_Float32),
+                       'los_north': (
```
Comment (LiangJYu) [inline src/compass/s1_rdr2geo.py:122]: `los_east` and `los_north` (as well other topo layer file names) are hard coded here and in `s1_geoocde_metadata.py`. What do you think about consolidating the files names into one place that can be imported from `rdr2geo.py`? e.g. Do the following to `s1_rdr2geo.py`:
```python
# list of file names for rasters created by rdr2geo
...
file_name_los_east = 'los_east'
file_name_los_north = 'los_north'

def run():
...
    topo_output = {
    file_name_los_east: (rdr2geo_cfg.compute_ground_to_sat_east, gdal.GDT_Float32),
...
}
```
Then in `s1_geocode_metadata.py` do the following:
```python
from rdr2geo import file_name_los_east, ...
...
def run():
    ...
    static_layers = {
        ...
        'east_los_vector': (cfg.rdr2geo_params.compute_ground_to_sat_east,
                            file_name_los_east),
        ...
    }
```

File: src/compass/s1_geocode_metadata.py [hunk 3]
```diff
@@ -138,10 +138,10 @@ def run(cfg, burst, fetch_from_scratch=False):
...
          'local_incidence_angle': (cfg.rdr2geo_params.compute_local_incidence_angle,
                                    'local_incidence'),
-         'heading_angle': (cfg.rdr2geo_params.compute_azimuth_angle,
-                           'heading'),
-         'layover_shadow_mask': (cfg.rdr2geo_params.compute_layover_shadow_mask,
-                                 'layover_shadow_mask')
+         'east_los_vector': (cfg.rdr2geo_params.compute_ground_to_sat_east,
+                            'los_east'),
+         'north_los_vector': (cfg.rdr2geo_params.compute_ground_to_sat_north,
+                              'los_north')
```
Comment (LiangJYu) [inline src/compass/s1_geocode_metadata.py:144]: See comment below about having single layer file name in `s1_rdr2geo.py`.

File: src/compass/schemas/s1_cslc_geo.yaml [hunk 4]
```diff
@@ -127,6 +127,10 @@ rdr2geo_options:
     compute_local_incidence_angle: bool(required=False)
     # Enable/disable azimuth angle output
     compute_azimuth_angle: bool(required=False)
+    # Enable/disable ground to satellite East LOS vector
+    compute_ground_to_sat_east: bool(required=False)
+    # Enable/disable ground to satellite North LOS vector
+    compute_ground_to_sat_north: bool(required=False)
```
Comment (LiangJYu) [inline src/compass/schemas/s1_cslc_geo.yaml:133]: These need to be added to defaults.

Comment (vbrancat) [reply src/compass/schemas/s1_cslc_geo.yaml:129]: They have been already added to the defaults. Please, double-check.

**Comment by vbrancat** [comment]
No sure why the measure of the runtime is off when I ran the static layers:

```
journal: elapsed time (GEO-IN) [s]: 6.477
journal: s1_geocode_metadata burst successfully ran in 0:00:46 (hr:min:sec)
journal: s1_static_layers burst successfully ran in 19066 days, 15:54:45 (hr:min:sec)
```

Any idea how to fix this @LiangJYu ?

**Comment by LiangJYu** [comment]
> No sure why the measure of the runtime is off when I ran the static layers:
> 
> ```
> journal: elapsed time (GEO-IN) [s]: 6.477
> journal: s1_geocode_metadata burst successfully ran in 0:00:46 (hr:min:sec)
> journal: s1_static_layers burst successfully ran in 19066 days, 15:54:45 (hr:min:sec)
> ```
> 
> Any idea how to fix this @LiangJYu ?

This [line](https://github.com/vbrancat/COMPASS/blob/48b63c10b6dcd847fd400dac384f75261ed75e5c/src/compass/s1_static_layers.py#L73) should look like this [one](https://github.com/vbrancat/COMPASS/blob/48b63c10b6dcd847fd400dac384f75261ed75e5c/src/compass/s1_geocode_metadata.py#L217).

**Comment by vbrancat** [comment]
@seongsujeong can you please review this PR? Thanks

File: src/compass/s1_cslc_qa.py [hunk 5]
```diff
@@ -127,12 +131,12 @@ def compute_static_layer_stats(self, cslc_h5py_root, rdr2geo_params):
 
         # Get the static layer to compute stats for
         static_layers_dict = {
-            'x': rdr2geo_params.compute_longitude,
-            'y': rdr2geo_params.compute_latitude,
-            'z': rdr2geo_params.compute_height,
-            'incidence_angle': rdr2geo_params.compute_incidence_angle,
-            'local_incidence_angle': rdr2geo_params.compute_local_incidence_angle,
-            'heading_angle': rdr2geo_params.compute_azimuth_angle
+            file_name_x: rdr2geo_params.compute_longitude,
```
Comment (LiangJYu) [inline src/compass/s1_cslc_qa.py:134]: nit - Can you expand the comment above to explicitly describe key and values.

**Review by LiangJYu — APPROVED** [review]
LGTM. Just one small nit.

**Comment by LiangJYu** [comment]
> > No sure why the measure of the runtime is off when I ran the static layers:
> > ```
> > journal: elapsed time (GEO-IN) [s]: 6.477
> > journal: s1_geocode_metadata burst successfully ran in 0:00:46 (hr:min:sec)
> > journal: s1_static_layers burst successfully ran in 19066 days, 15:54:45 (hr:min:sec)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Any idea how to fix this @LiangJYu ?
> 
> This [line](https://github.com/vbrancat/COMPASS/blob/48b63c10b6dcd847fd400dac384f75261ed75e5c/src/compass/s1_static_layers.py#L73) should look like this [one](https://github.com/vbrancat/COMPASS/blob/48b63c10b6dcd847fd400dac384f75261ed75e5c/src/compass/s1_geocode_metadata.py#L217).

The 2 following places need `time.time()` changed to `time.perf_counter()`
[s1_geo2rdr.py](https://github.com/opera-adt/COMPASS/blob/48b63c10b6dcd847fd400dac384f75261ed75e5c/src/compass/s1_geo2rdr.py#L31)
[s1_resample.py](https://github.com/opera-adt/COMPASS/blob/48b63c10b6dcd847fd400dac384f75261ed75e5c/src/compass/s1_resample.py#L31)

File: src/compass/s1_cslc_qa.py [hunk 6]
```diff
@@ -8,6 +8,10 @@
 import isce3
 import numpy as np
 
+from compass.s1_rdr2geo import (file_name_los_east,
+                                file_name_los_north,file_name_local_incidence,
+                                file_name_layover, file_name_x,
```
Comment (seongsujeong) [inline src/compass/s1_cslc_qa.py:13]: Looks like `file_name_layover` is not used in this python code.

File: src/compass/s1_geocode_metadata.py [hunk 7]
```diff
@@ -12,6 +12,10 @@
 from scipy.interpolate import InterpolatedUnivariateSpline
 
 from compass import s1_rdr2geo
+from compass.s1_rdr2geo import (file_name_los_east,
+                                file_name_los_north,file_name_local_incidence,
```
Comment (seongsujeong) [inline src/compass/s1_geocode_metadata.py:16]:
```suggestion
                                file_name_los_north, file_name_local_incidence,
```

File: src/compass/s1_rdr2geo.py [hunk 8]
```diff
@@ -103,37 +111,40 @@ def run(cfg, burst=None, save_in_scratch=False):
...
             isce3.io.Raster(f'{output_path}/{fname}.rdr', rdr_grid.width,
                             rdr_grid.length, 1, dtype, 'ENVI')
             if enabled else None
             for fname, (enabled, dtype) in topo_output.items()]
 
         x_raster, y_raster, z_raster, layover_shadow_raster, \
-        incident_angle_raster, local_incident_angle_raster, \
-        azimuth_angle_raster = raster_list
+        local_incident_angle_raster, los_east_raster,\
+        los_north_raster = raster_list
```
Comment (seongsujeong) [inline src/compass/s1_rdr2geo.py:139]: Just a style suggestion. You can keep the current style line splitting if you want.
```python
        (x_raster, y_raster, z_raster, layover_shadow_raster,
        local_incident_angle_raster, los_east_raster,
        los_north_raster) = raster_list
```

**Review by seongsujeong — COMMENTED** [review]
Looks good overall. My comments are as below - Seongsu

**Comment by LiangJYu** [comment]
Process timing counter fixed and @seongsujeong comments addressed.

**Review by seongsujeong — APPROVED** [review]
LGTM.

**Pull request merged**